### ISSUE TITLE
How to choose weights in PINNs? Is there anything implemented in DeepXDE in this direction?

### ISSUE THREAD

**Issue by fperiago** [lululxvi/deepxde#1953]
Hello all,

As you all know, the selection of the weights for the different contributions in the loss function is  a crucial issue to the success of the algo. 
I wonder whether there is something implemented in DeepXDE in this direction. 

Thank you in advance.

**Comment by kyouma** [comment]
You could do `model.loss_weights = [new, values, for, all, losses]` after `model.train(iterations=N, ...)`, where `N < whole number of iterations`, with new weights each N steps (perhaps, each 1 step may produce too large overhead, but you could try) and set the weights according to one of these articles (the 1st one is simpler, see _Algorithm 1_):

> - Wang, S., Sankaran, S., Wang, H. and Perdikaris, P., 2023. An expert's guide to training physics-informed neural networks. arXiv preprint arXiv:2308.08468.
> - Xiang, Z., Peng, W., Zheng, X., Zhao, X. and Yao, W., 2021. Self-adaptive loss balanced Physics-informed neural networks for the incompressible Navier-Stokes equations. arXiv preprint arXiv:2104.06217.
> - Xiang, Z., Peng, W., Liu, X. and Yao, W., 2022. Self-adaptive loss balanced physics-informed neural networks. Neurocomputing, 496, pp.11-34.
> - Hua, J., Li, Y., Liu, C., Wan, P. and Liu, X., 2023. Physics-informed neural networks with weighted losses by uncertainty evaluation for accurate and stable prediction of manufacturing systems. IEEE Transactions on Neural Networks and Learning Systems.

You can obtain the last training loss values from either `train_state.loss_train`, `losshistory.loss_train[-1]` or `model.losshistory.loss_train[-1]`. I am not sure if they differ in some conditions, but they seem to produce the same results, even if the `iterations` parameter of `model.train()` can not be divided by `display_every`.

**Comment by fperiago** [comment]
Thank you for your help
